src/flow_graph.py

```diff
@@ -917,20 +921,27 @@ def find_block_by_label(label: str) -> Optional[Block]:
                 )
 
             jtbl_name = jtbl_names[0]
-            if jtbl_name not in rodata.values:
+            if jtbl_name not in data_section.values:
                 raise DecompFailure(
                     f"Found jr instruction {jump.meta.loc_str()}, but the "
                     "corresponding jump table is not provided.\n"
                     "\n"
                     "Please include it in the input .s file, or in a separate .s "
-                    "file pointed to by --rodata.\n"
+                    "file pointed to by --data.\n"
                     'It needs to be within ".section .rodata" or ".section .late_rodata".\n'
                     "\n"
                     "(You might need to pass --goto and --no-andor flags as well, "
                     "to get correct control flow for non-jtbl switch jumps.)"
                 )
 
-            jtbl_entries = rodata.values[jtbl_name].data
+            jtbl_data_entry = data_section.values[jtbl_name]
+            if not jtbl_data_entry.is_readonly:
```

Comment:
this seems like an unnecessary check to me, I could see decompiling code where sections are improperly annotated

Reply:
I was kind of worried about emitting incorrect output after assuming that a jumptable was read-only -- but you're right, I think that's way less common than improperly-annotated code.